### ISSUE TITLE
Support for subclassing midi::Message for use for MIDI SysEx

### ISSUE THREAD

**Issue by stoermelder** [VCVRack/Rack#1433]
### Summary

I'm working on a module and trying to send Midi SysEx. I believe it is currently not possible, because the fields `bytes` and `size` of `midi::Message` are directly accessed in the driver and cannot be "overriden" by a subclass of `midi::Message`.

https://github.com/VCVRack/Rack/blob/266b63d5aa7a952ce17b2d6170d51549fcdbd6c9/src/rtmidi.cpp#L71

### Proposed design

Access `bytes` and `size` not directly in the midi driver. Instead add and use methods like `virtual uint8_t *midi::Message::getBytes()` and `virtual uint8_t midi::Message::getSize()`.

If I'm wrong and it is possible please let me know.

**Comment by AndrewBelt** [comment]
This API can't change until v2.

**Comment by alasdairmoons** [comment]
Please include a possibility to skipping the InputQueue for Midi System Messages (message->size == 1) to access them in Realtime from the Inputstream for enhancement of the 0xF8 - 0xFC jitters/latency, if this is not already included in the request above.

**Comment by AndrewBelt** [comment]
Sure, just use `Input` instead of `InputQueue`.

Why again does `MidiMessage` need to change for SysEx? Do you need MIDI messages more than 3 bytes?

**Comment by alasdairmoons** [comment]
> Sure, just use `Input` instead of `InputQueue`.
Perfect, thanks @AndrewBelt 

> Do you need MIDI messages more than 3 bytes?
Not me, 1 Status- and 2 Databytes like Midistandard.

**Comment by AndrewBelt** [comment]
Are you sending or receiving MIDI? If sending, just set `msg.setSize(1)` etc. If receiving, inspecting `msg.size` will be 1 or 2.

**Comment by AndrewBelt** [comment]
Closing as already-implemented.

**Comment by stoermelder** [comment]
> Why again does `MidiMessage` need to change for SysEx? Do you need MIDI messages more than 3 bytes?

This was my request, please re-open. 
Yes, most SysEx messages need more than 3 bytes.

**Comment by AndrewBelt** [comment]
Made `midi::Message` a wrapper for `std::vector<uint8_t> bytes;` in Rack v2, initialized to 3 empty bytes.

**Comment by wipu** [comment]
Heads up: I just wrote a fork that _adds_ support for longer (sysex) sends to v1: https://github.com/wipu/Rack/tree/sysex-send

I'm using my fork for my Commodore 64 controller plugin: https://github.com/wipu/sidex-vcv

The underlying dependency, rtmidi, already supports what I need so I just needed to add some trivial shoveling of data to the VCV api. I don't know what the limitations are, but at least in my usecase (messages of 35 bytes) things seem to work just ok.

Since _adding_ functions almost never breaks compatibility I don't see why my patch (or a cleaner and coding-convention-obeying version with "use if you know what you are doing" comments) couldn't be add in a "just do it" fashion.
